presto-main/src/main/resources/webapp/src/components/QueryList.jsx

```diff
@@ -469,7 +469,7 @@ export class QueryList extends React.Component {
             newSortOrder = SORT_ORDER.ASCENDING;
         }
 
-        const newDisplayedQueries = this.filterQueries(this.state.allQueries, this.state.stateFilters, this.state.searchString);
```

Comment:
I see now when it happens. For example when you update the sort filter, such as ordering by *Elapsed Time*
<img width="1112" alt="screen shot 2018-11-01 at 9 34 19 pm" src="https://user-images.githubusercontent.com/190738/47894105-253dea80-de1e-11e8-92d8-e98fb15a8c18.png">. The missing argument of `filterQueries()` ends up with the `undefined` value.

Should we use TypeScript and Flow eventually to avoid such errors?

Reply:
It is a good example of a fix that it would be nice to deploy independently of the usual Presto "core" release cycle.

Reply:
100% agree that we need better type safety - all the JSON stuff is super brittle. Also agree that it would be nice if we can deploy this independently.